libs/cart/src/reducers/cart/cart.reducer.spec.ts

```diff
@@ -0,0 +1,286 @@
+      expect(result.cart.id).toEqual(cart.id)
+    });
+
+    it('should setloading to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+  });
+
+  describe('when CartCreateFailureAction is triggered', () => {
+    const error = 'error message';
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      state = {
+        ...initialState,
+        loading: true,
+        errors: new Array('firstError')
+      }
+
+      const cartCreateFailure = new DaffCartCreateFailure(error);
+
+      result = reducer(state, cartCreateFailure);
+    });
+
+    it('should setloading to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+
+    it('adds an error to state.errors', () => {
+      expect(result.errors.length).toEqual(2);
+    });
+  });
+
+  describe('when AddToCartAction is triggered', () => {
+    const productId = 'productId';
+    const qty = 1;
+
+    it('should setloading state to true', () => {
+      const addToCartAction: DaffAddToCart = new DaffAddToCart({ productId, qty });
+
+      const result = reducer(initialState, addToCartAction);
+
+      expect(result.loading).toEqual(true);
+    });
+  });
+
+  describe('when AddToCartActionSuccess is triggered', () => {
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      const addToCartActionSuccess: DaffAddToCartSuccess = new DaffAddToCartSuccess(cart);
+      state = {
+        ...initialState,
+        loading: true
+      }
+
+      result = reducer(state, addToCartActionSuccess);
+    });
+
+    it('should setcart from action.payload', () => {
+      expect(result.cart).toEqual(cart)
+    });
+
+    it('should setloading state to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+  });
+
+  describe('when AddToCartFailureAction is triggered', () => {
+    let error: string;
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      state = {
+        ...initialState,
+        loading: true,
+        errors: new Array('firstError')
+      }
+
+      error = 'error';
+
+      const addToCartFailure = new DaffAddToCartFailure(error);
+
+      result = reducer(state, addToCartFailure);
+    });
+
+    it('should setloading to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+
+    it('adds an error to state.errors', () => {
+      expect(result.errors.length).toEqual(2);
+    });
+  });
+
+  describe('when CartClearAction is triggered', () => {
+    it('should indicate that the cart is loading', () => {
+      const expectedState = {
+        loading: true,
+        cart: null,
+        errors: []
+      }
+      const cartReset = new DaffCartClear();
+      const result = reducer(initialState, cartReset);
+
+      expect(result).toEqual(expectedState);
+    });
+  });
+
+  describe('when CartClearSuccessAction is triggered', () => {
+    it('should guarantee there are no items in the cart', () => {
```

Comment:
The text here should probably be something like: "should set the cart payload on state". I don't see anything in this test about guaranteeing that there are no items, because that happens in the driver.

Reply:
Changed in [`4dcc86c`](https://github.com/graycoreio/daffodil/pull/681/commits/4dcc86c5406fda6bab2ad536c6f3bdbf8e4d2111).